test/widgets/action_sheet_test.dart

```diff
@@ -553,6 +553,21 @@ void main() {
     });
   });
 
+
+  group('narrowLink URL generation tests', () {
+
+    testWidgets('Full Url Testing', (tester) async {
+      final message = eg.streamMessage();
+      final narrow = TopicNarrow.ofMessage(message);
+      await setupToMessageActionSheet(tester, message: message, narrow: narrow);
+      final store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+      final generatedLink = narrowLink(store, narrow, nearMessageId: message.id).toString();
```

Comment:
To make this end-to-end, let's generate the link through `tapCopyMessageLinkButton`. We should move this test to the "CopyMessageLinkButton" group.